dataplanes/vpp/pkg/nsmdataplane/nsmdataplane_server.go

```diff
@@ -57,13 +66,14 @@ func createLocalConnect(apiCh govppapi.Channel, src, dst *common.LocalMechanism)
 		return "", err
 	}
 
-	return nsmvpp.CreateLocalConnect(apiCh, src.Parameters, dst.Parameters)
+	return mechanisms[src.Type].CreateLocalConnect(apiCh, src.Parameters, dst.Parameters)
 }
 
 // deleteLocalConnect sanity checks parameters passed in the LocalMechanisms and call nsmvpp.CreateLocalConnect
 func deleteLocalConnect(apiCh govppapi.Channel, connID string) error {
+	mechanism, _ := strconv.Atoi(strings.Split(connID, "-")[0])
```

Comment:
I am a little concerned about deriving mechanism type from parsing the connID.  Is there a place where the vpp dataplane tracks its current connections indexed by connID such that it can look up the Connection object from the connID and get the mechanism from there?

Reply:
It would be better to do as you suggested, as far as I know there is no place yet where vpp dataplane tracks current connections, let's take mechanism from there once we'll have such place.